gate-web/src/main/groovy/com/netflix/spinnaker/gate/config/SlackConfig.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.config;
+
+import static retrofit.Endpoints.newFixedEndpoint;
+
+import com.netflix.spinnaker.gate.services.SlackService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import retrofit.Endpoint;
+import retrofit.RequestInterceptor;
+import retrofit.RestAdapter;
+import retrofit.client.OkClient;
+import retrofit.converter.JacksonConverter;
+
+@Configuration
+@EnableConfigurationProperties(SlackConfigProperties.class)
+public class SlackConfig {
+
+  private SlackConfigProperties slackConfigProperties;
+
+  @Autowired
+  public SlackConfig(SlackConfigProperties slackConfigProperties) {
+    this.slackConfigProperties = slackConfigProperties;
+  }
+
+  @Bean
+  Endpoint slackEndpoint() {
+    return newFixedEndpoint(slackConfigProperties.baseUrl);
+  }
+
+  RequestInterceptor requestInterceptor =
+      new RequestInterceptor() {
+        @Override
+        public void intercept(RequestInterceptor.RequestFacade request) {
+          String value = "Token token=" + slackConfigProperties.token;
+          request.addHeader("Authorization", value);
+        }
+      };
+
+  @Bean
+  SlackService slackService(Endpoint slackEndpoint) {
+    return new RestAdapter.Builder()
+        .setEndpoint(slackEndpoint)
+        .setClient(new OkClient())
```

Comment:
You can use ok http3 client here:
 ``` 
@Bean
  Ok3Client okClient(OkHttp3ClientConfiguration okHttpClientConfig) {
    return new Ok3Client(okHttpClientConfig.create().build());
  }
```

And pass it as a param to:
```
 @Bean
  SlackService slackService(Endpoint slackEndpoint, Ok3Client okClient) ...
```

Reply:
I initialized the http3 client a bit differently because the config is causing an error since this request should be anonymous.

Reply:
There are probably ssl errors? it looks like `com.netflix.spinnaker.config.OkHttp3ClientConfiguration`  is more suitable for ipc between spinnaker services. Please leave as is.